include/xleaflet/xwidget_control.hpp

```diff
@@ -0,0 +1,134 @@
+/******************************************************************************
+* Copyright (c) 2018, Sylvain Corlay and Johan Mabille, Wolf Vollprecht and   *
+* Martin Renou                                                                *
+*                                                                             *
+* Distributed under the terms of the BSD 3-Clause License.                    *
+*                                                                             *
+* The full license is in the file LICENSE, distributed with this software.    *
+*******************************************************************************/
+
+#ifndef XLEAFLET_WIDGET_CONTROL_HPP
+#define XLEAFLET_WIDGET_CONTROL_HPP
+
+#include <string>
+
+#include "xwidgets/xholder.hpp"
+#include "xwidgets/xmaterialize.hpp"
+#include "xwidgets/xwidget.hpp"
+
+#include "xcontrol.hpp"
+
+namespace xlf
+{
+    /*********************
+     * widget_control declaration *
+     *********************/
+
+    template <class D>
+    class xwidget_control : public xcontrol<D>
+    {
+    public:
+
+        using point_type = std::array<double, 2>;
+
+        using base_type = xcontrol<D>;
+        using derived_type = D;
+
+        using widget_type = xw::xholder<xw::xwidget>;
+
+        void serialize_state(xeus::xjson&, xeus::buffer_sequence&) const;
+        void apply_patch(const xeus::xjson&, const xeus::buffer_sequence&);
+
+        XPROPERTY(xtl::xoptional<widget_type>, derived_type, widget);
+        XPROPERTY(int, derived_type, max_width, 300);
```

Comment:
In ipyleaflet it's
```python
max_width = Int(default_value=None, allow_none=True).tag(sync=True)
```
So that would be an `xoptional` here with a default value of `missing` I guess.